### ISSUE TITLE
Closely Implement Java Standard Library

### ISSUE THREAD

**Issue by Yevano** [Team-CC-Corp/JVML-JIT#25]
This is one of those issues which is gonna stay an issue for a really, really long time.

I think one of our goals should be to be able to run actual already created Java programs and libraries with little to no porting involved. Namely, the Java compiler would be extremely useful to be able to run within ComputerCraft, as it would mean we can actually write Java programs on servers and such where we don't have access to the external Java compiler.

Right now, most of the included classes have parts pruned as a matter of simplification. I think the approach we should take should be fairly relaxed. If we add new features which allows certain parts of classes to be implemented, we should try our best to implement those classes completely until we can do a direct copy-paste from the Java source and it just works.

We don't really need to worry about this too much until the actual VM is completely implemented. (Which is not too far down the road I think.) After that, the only obstacle is implementing the native methods which the library will depend on. Just looking at the Java source, the native runtime stuff is pretty intricate, so this will be somewhat difficult. I think that it might not be too bad to simplify the native backbone where we can in favor of getting all the rest of the actual Java stuff working, and then go back later and fill it out just for consistency.

tl;dr we should implement the standard library but vm and natives are the main obstacles so we should take it one step at a time.
